graylog2-web-interface/src/components/extractors/MessageLoader.tsx

```diff
@@ -0,0 +1,134 @@
+
+import ActionsProvider from 'injection/ActionsProvider';
+
+import { Button } from '../graylog';
+
+const MessagesActions = ActionsProvider.getActions('Messages');
+
+type LoadMessageFormProps = {
+  loadMessage: (e: React.FormEvent) => void,
+  children: React.ReactNode,
+  loading: boolean,
+};
+const LoadMessageForm = ({ loadMessage, children, loading }: LoadMessageFormProps) => (
+  <div>
+    <form className="form-inline message-loader-form" onSubmit={loadMessage}>
+      {children}
+      <Button bsStyle="info"
+              disabled={loading}
+              type="submit">
+        {loading ? 'Loading message...' : 'Load message'}
+      </Button>
+    </form>
+  </div>
+);
+
+type Props = {
+  hidden: boolean,
+  hideText: string,
+  onMessageLoaded: (data: any) => void,
+  messageId: undefined | string,
+  index: undefined | string,
+};
+
+const useMessageLoader = (defaultMessageId: string, defaultIndex: string, onMessageLoaded: (data: any) => void): [boolean, (messageId: string, index: string) => void] => {
+  const [loading, setLoading] = useState(false);
+  const loadMessage = useCallback((messageId: string, index: string) => {
+    if (messageId === '' || index === '') {
+      return;
+    }
+
+    setLoading(true);
+    const promise = MessagesActions.loadMessage(index, messageId);
+
+    promise.then((data) => onMessageLoaded(data));
+    promise.finally(() => setLoading(false));
+  }, [onMessageLoaded]);
+
+  useEffect(() => {
+    if (defaultMessageId && defaultIndex) {
+      loadMessage(defaultMessageId, defaultIndex);
+    }
+  }, [defaultMessageId, defaultIndex, loadMessage]);
+
+  return [loading, loadMessage];
+};
+
+const MessageLoader = ({ hidden, hideText, onMessageLoaded, messageId: defaultMessageId, index: defaultIndex }: Props) => {
+  const [isHidden, setIsHidden] = useState(hidden);
+
+  const [messageId, setMessageId] = useState<string>(defaultMessageId);
+  const onChangeMessageId = useCallback((e) => setMessageId(e.target.value), []);
+  const messageIdRef = useRef<HTMLInputElement>();
+
+  const [index, setIndex] = useState<string>(defaultIndex);
+  const onChangeIndex = useCallback((e) => setIndex(e.target.value), []);
+
+  const [loading, loadMessage] = useMessageLoader(defaultMessageId, defaultIndex, onMessageLoaded);
+
+  const _loadMessage = useCallback((e?: React.FormEvent) => {
+    if (e?.preventDefault) {
+      e.preventDefault();
+    }
+
+    loadMessage(messageId, index);
+  }, [index, loadMessage, messageId]);
+
+  const toggleMessageForm = useCallback(() => {
+    const newIsHidden = !isHidden;
+    setIsHidden(newIsHidden);
+
+    if (!newIsHidden) {
+      messageIdRef.current?.focus?.();
+    }
+  }, [isHidden]);
+
+  return (
+    <div className="message-loader">
+      {hideText || (
+        <p>
+          Wrong example? You can{' '}
+          <Button bsStyle="link" bsSize="sm" onClick={toggleMessageForm}>load another message</Button>.
+        </p>
+      )}
+      {hidden || (
```

Comment:
Using `hidden` will not allow users to display the form when clicking on "Load another message", I think this line should use `isHidden` instead:

```suggestion
      {isHidden || (
```

Reply:
:white_check_mark: